packages/compass-utils/src/filesystem.ts

```diff
@@ -0,0 +1,145 @@
+import { promises as fs, createWriteStream } from 'fs';
+import path from 'path';
+import { glob } from 'glob';
+import { createLoggerAndTelemetry } from '@mongodb-js/compass-logging';
+import { getAppPath } from './electron';
+
+const { log, mongoLogId } = createLoggerAndTelemetry('COMPASS-UTILS');
+
+type FilesystemOptions<T = unknown> = {
+  subdir: string;
+  onSerialize: (content: T) => string;
+  onDeserialize: (content: string) => T;
+};
+
+export class Filesystem<T> {
+  private readonly subdir: string;
+  private readonly onSerialize: FilesystemOptions<T>['onSerialize'];
+  private readonly onDeserialize: FilesystemOptions<T>['onDeserialize'];
+
+  constructor({
+    subdir = '/',
+    onSerialize = (content: T) => JSON.stringify(content, null, 2),
+    onDeserialize = JSON.parse,
+  }: Partial<FilesystemOptions<T>> = {}) {
+    this.subdir = subdir;
+    this.onDeserialize = onDeserialize;
+    this.onSerialize = onSerialize;
+  }
+
+  private getStorageBasePath(): string {
+    if (process.env.COMPASS_TESTS_STORAGE_BASE_PATH) {
+      return path.join(
+        process.env.COMPASS_TESTS_STORAGE_BASE_PATH,
+        this.subdir
+      );
+    }
+    const basepath = getAppPath();
+    if (!basepath) {
+      throw new Error('The storage path is not defined.');
```

Comment:
I guess this is more like "Using storage in non-supported environment". The way we are changing how it is used though, I think it should be fine if `getAppPath` will throw now instead of silently not returning anything

Reply:
just a thought, i think that throwing `'The storage path is not defined.'` is less confusing than trying to derive a cause. Maybe a combo of both? `The storage path is not defined. Using storage in non-supported environment?`